statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,502 @@
+            exposure = getattr(self, 'exposure', 0)
+
+        if exog_infl is None:
+            exog_infl = self.exog_infl
+
+        if exposure is None:
+            exposure = 0
+        elif exposure != 0:
+            exposure = np.log(exposure)
+
+        if offset is None:
+            offset = 0
+
+        params_infl = params[:self.k_inflate]
+        params_main = params[self.k_inflate:]
+
+        if self.infl == 'logit':
+            prob_poisson = 1 / (1 + np.exp(np.dot(exog_infl, params_infl)))
+        elif self.infl == 'probit':
+            raise NotImplemented('Predict for Probit inflation not implemented')
+            
+
+        lin_pred = np.dot(exog, params_main) + exposure + offset
+        prob_zero = ((1 - prob_poisson) +
+            prob_poisson * np.exp(self.model_main.loglikeobs(params_main)))
+
+        if which == 'mean':
+            return prob_poisson * np.exp(lin_pred)
+        elif which == 'poisson-mean':
+            return np.exp(lin_pred)
+        elif which == 'linear':
+            return lin_pred
+        elif which == 'mean-nonzero':
+            return prob_poisson * np.exp(lin_pred) / (1 - prob_zero)
+        elif which == 'prob-zero':
+            return  prob_zero
+        elif which == 'prob':
+            counts = np.atleast_2d(np.arange(0, np.max(self.endog)+1))
+            w = self.model_infl.predict(params_infl)[:, None]
+            w[w == 1.] = np.nextafter(1, 0)
+            mu = self.model_main.predict(params_main)[:, None]
+            return self.distribution.pmf(counts, mu, w)
+        else:
+            raise ValueError('keyword `which` not recognized')
+
+class PoissonZeroInflated(GenericZeroInflated):
+    """
+    Poisson Zero Inflated model for count data
+
+    %(params)s
+    %(extra_params)s
+
+    Attributes
+    -----------
+    endog : array
+        A reference to the endogenous response variable
+    exog : array
+        A reference to the exogenous design.
+    exog_infl: array
+        A reference to the zero-inflated exogenous design.
+    """ % {'params' : base._model_params_doc,
+           'extra_params' :
+           """offset : array_like
+        Offset is added to the linear prediction with coefficient equal to 1.
+    exposure : array_like
+        Log(exposure) is added to the linear prediction with coefficient
+        equal to 1.
+
+    """ + base._missing_param_doc}
+
+    def __init__(self, endog, exog, exog_infl=None, offset=None, exposure=None,
+                 inflation='logit', missing='none', **kwargs):
+        super(PoissonZeroInflated, self).__init__(endog, exog, offset=offset,
+                                                  inflation=inflation,
+                                                  exog_infl=exog_infl,
+                                                  exposure=exposure,
+                                                  missing=missing, **kwargs)
+        self.model_main = Poisson(self.endog, self.exog)
```

Comment:
I guess that this should include offset and exposure.
We need unit tests that cover the case when offset and exposure are provided by the user.